src/vm/ceemain.cpp

```diff
@@ -3401,19 +3401,13 @@ void ContractRegressionCheck()
 
 
 //
-// GetOSVersion - Gets the real OS version bypassing the OS compatibility shim
-// Mscoree.dll resides in System32 dir and is always excluded from compat shim.
-// This function calls mscoree!shim function via mscoreei ICLRRuntimeHostInternal interface
-// to get the OS version. We do not do this PAL or coreclr..we direclty call the OS
-// in that case.
+// GetOSVersion - OS does not provide apis to get the OSVersion
 //
 BOOL GetOSVersion(LPOSVERSIONINFO lposVer)
 {
-// Fix for warnings when building against WinBlue build 9444.0.130614-1739
-// warning C4996: 'GetVersionExW': was declared deprecated
-// externalapis\windows\winblue\sdk\inc\sysinfoapi.h(442)
-// Deprecated. Use VerifyVersionInfo* or IsWindows* macros from VersionHelpers.
-#pragma warning( disable : 4996 )
-    return WszGetVersionEx(lposVer);
-#pragma warning( default : 4996 )
+    if (lposVer == nullptr)
+        return FALSE;
+    memset(lposVer, 0, sizeof(OSVERSIONINFO));
+    lposVer->dwPlatformId = VER_PLATFORM_WIN32_NT;
```

Comment:
How about Unix? We have GetVersionEx implementation in PAL that sets dwPlatformId to VER_PLATFORM_UNIX.
And if we don't want to call the PAL, could you please remove the GetVersionEx implementation from PAL and also remove the PAL tests that excercise it?
Also, the `#define WszGetVersionEx GetVersionExW` should be removed from winwrap.h in this case.